File: 프로그래머스/2/60057. 문자열 압축/문자열 압축.py
import sys
def solution(s):
    answer = sys.maxsize
    
    # i는 문자를 자르는 단위
    for i in range(1, len(s)+1):
        b = ''
        tmp = s[:i]
        cnt = 1
        # 가장 짧은 문자열 찾기
        for j in range(i,len(s)+i, i):
            if tmp == s[j:j+i]:
                cnt +=1
            else:
                if cnt != 1:
                    b += str(cnt) + tmp
                else:
                    b+= tmp
                # 기준 문자 바꾸기
                tmp = s[j:j+i]
                cnt = 1
        answer = min(answer, len(b))
    return answer

File: 프로그래머스/2/60057. 문자열 압축/문자열 압축.java
import java.util.*;

class Solution {
    public int solution(String s) {
        int answer = s.length();
        
        for(int i = 1; i <= s.length(); i++) {
            String target = s.substring(0, i);
            String result = "";
            int count = 1;
            
            for (int j = i; j <= s.length(); j= j + i) {
                int endIndex = Math.min(j+i, s.length());
                String temp = s.substring(j, endIndex);
                
                if (target.equals(temp)){
                    count++;
                } else{
                    if (count > 1){
                        result += String.valueOf(count);
                    }
                    result += target;
                    target = temp;
                    count = 1;
                }
            }
            result += target;
            answer = Math.min(answer, result.length());
        }
        return answer;
    }
}